### PR TITLE
misc. work on coach tables

### PR DESCRIPTION
**Pull request by indirectlylit**
some minor refactoring and clean-up

### CODE REVIEW

**Comment by indirectlylit** [comment]
merging in to keep things current - feel free to add comments

File: kolibri/plugins/coach_tools/assets/src/vue/reports/index.vue [hunk 1]
```diff
@@ -155,38 +157,15 @@
...
-        };
-        Object.assign(vlink.params, currentParams, newParams);
-        return vlink;
+        return genLink(this.pageState, { all_or_recent: Constants.AllOrRecent.ALL });
       },
-      switchView(isContent) {
-        const newView = isContent ? Constants.ViewBy.CONTENT : Constants.ViewBy.LEARNERS;
-        this.$router.go(this.genLink({ view_by_content_or_learners: newView }));
+      viewByLink() {
+        const view = this.isViewByContent ? Constants.ViewBy.LEARNERS : Constants.ViewBy.CONTENT;
```
Comment (christianmemije) [inline kolibri/plugins/coach_tools/assets/src/vue/reports/index.vue:163]: Should the expressions be switched?

Comment (indirectlylit) [reply kolibri/plugins/coach_tools/assets/src/vue/reports/index.vue:163]: no - target of the link is the opposite of the current view. I'll add a clarifying comment

Comment (christianmemije) [reply kolibri/plugins/coach_tools/assets/src/vue/reports/index.vue:163]: Make sense. Thanks.

**Pull request merged**